src/Flysystem/FlysystemAssetStore.php

```diff
@@ -318,6 +317,53 @@ public function delete($filename, $hash)
         return $protected || $public;
     }
 
+    public function rename($filename, $hash, $newName)
+    {
+        return $this->copyOrRename($filename, $hash, $newName, true);
+    }
+
+    public function copy($filename, $hash, $newName)
+    {
+        return $this->copyOrRename($filename, $hash, $newName, false);
+    }
+
+    /**
+     * Copy or move a file and all variants
+     *
+     * @param string $filename urrent filename
+     * @param string $hash Hash
+     * @param string $newName filename
+     * @param bool $asRename Set to true to rename (instead of copy)
```

Comment:
Boolean flag to switch between 2 different methods is a bit yuck. :-P

This might be better private than protected as even subclasses should use the copy & rename methods. That limits the impact of the less than ideal method signature.